File: Templates/java/MathTemplate.java
/*
 * Author  : Aritra Dutta
 * Target  : Codeforces Expert / CSES
 * 
 * MATH Template - Number Theory, Combinatorics, Primes
 * Uses optimized I/O for CSES acceptance
 */
import java.io.*;
import java.util.*;

public class MathTemplate {
    static BufferedReader br;
    static StringTokenizer st;
    static PrintWriter out;
    
    static final int MOD = 1_000_000_007;
    static final int MAXN = 2_000_001;
    
    static long[] fact = new long[MAXN];
    static long[] invFact = new long[MAXN];
    
    public static void main(String[] args) throws IOException {
        br = new BufferedReader(new InputStreamReader(System.in));
        out = new PrintWriter(new BufferedOutputStream(System.out));
        
        precomputeFactorials();
        
        int t = nextInt();
        while (t-- > 0) {
            solve();
        }
        
        out.flush();
        out.close();
    }
    
    static void solve() throws IOException {
        int n = nextInt(), r = nextInt();
        out.println(nCr(n, r));
    }
    
    // ==================== Modular Arithmetic ====================
    static long modAdd(long a, long b) {
        return ((a % MOD) + (b % MOD)) % MOD;
    }
    
    static long modSub(long a, long b) {
        return ((a % MOD) - (b % MOD) + MOD) % MOD;
    }
    
    static long modMul(long a, long b) {
        return ((a % MOD) * (b % MOD)) % MOD;
    }
    
    static long modPow(long x, long y, long m) {
        long res = 1;
        x %= m;
        while (y > 0) {
            if ((y & 1) == 1) res = res * x % m;
            x = x * x % m;
            y >>= 1;
        }
        return res;
    }
    
    static long modPow(long x, long y) {
        return modPow(x, y, MOD);
    }
    
    static long modInv(long x) {
        return modPow(x, MOD - 2, MOD);
    }
    
    // ==================== Factorials & nCr ====================
    static void precomputeFactorials() {
        fact[0] = 1;
        for (int i = 1; i < MAXN; i++) {
            fact[i] = modMul(fact[i - 1], i);
        }
        invFact[MAXN - 1] = modInv(fact[MAXN - 1]);
        for (int i = MAXN - 2; i >= 0; i--) {
            invFact[i] = modMul(invFact[i + 1], i + 1);
        }
    }
    
    static long nCr(int n, int r) {
        if (r < 0 || r > n) return 0;
        return modMul(fact[n], modMul(invFact[r], invFact[n - r]));
    }
    
    static long nPr(int n, int r) {
        if (r < 0 || r > n) return 0;
        return modMul(fact[n], invFact[n - r]);
    }
    
    // ==================== GCD, LCM ====================
    static long gcd(long a, long b) {
        return b == 0 ? a : gcd(b, a % b);
    }
    
    static long lcm(long a, long b) {
        return a / gcd(a, b) * b;
    }
    
    // ==================== Sieve of Eratosthenes ====================
    static boolean[] sieve(int n) {
        boolean[] isPrime = new boolean[n + 1];
        Arrays.fill(isPrime, true);
        isPrime[0] = isPrime[1] = false;
        
        for (int i = 2; i * i <= n; i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= n; j += i) {
                    isPrime[j] = false;
                }
            }
        }
        return isPrime;
    }
    
    static List<Integer> getPrimes(int n) {
        boolean[] isPrime = sieve(n);
        List<Integer> primes = new ArrayList<>();
        for (int i = 2; i <= n; i++) {
            if (isPrime[i]) primes.add(i);
        }
        return primes;
    }
    
    // ==================== Smallest Prime Factor ====================
    static int[] computeSPF(int n) {
        int[] spf = new int[n + 1];
        for (int i = 0; i <= n; i++) spf[i] = i;
        
        for (int i = 2; i * i <= n; i++) {
            if (spf[i] == i) {
                for (int j = i * i; j <= n; j += i) {
                    if (spf[j] == j) spf[j] = i;
                }
            }
        }
        return spf;
    }
    
    // Prime factorization using SPF - O(log n)
    static List<int[]> factorize(int n, int[] spf) {
        List<int[]> factors = new ArrayList<>();
        while (n > 1) {
            int p = spf[n], cnt = 0;
            while (n % p == 0) { n /= p; cnt++; }
            factors.add(new int[]{p, cnt});
        }
        return factors;
    }
    
    // ==================== Euler's Totient ====================
    static long phi(long n) {
        long result = n;
        for (long i = 2; i * i <= n; i++) {
            if (n % i == 0) {
                while (n % i == 0) n /= i;
                result -= result / i;
            }
        }
        if (n > 1) result -= result / n;
        return result;
    }
    
    // ==================== Matrix Exponentiation ====================
    static long[][] matMul(long[][] A, long[][] B) {
        int n = A.length;
        long[][] C = new long[n][n];
        for (int i = 0; i < n; i++) {
            for (int k = 0; k < n; k++) {
                for (int j = 0; j < n; j++) {
                    C[i][j] = modAdd(C[i][j], modMul(A[i][k], B[k][j]));
                }
            }
        }
        return C;
    }
    
    static long[][] matPow(long[][] A, long p) {
        int n = A.length;
        long[][] result = new long[n][n];
        for (int i = 0; i < n; i++) result[i][i] = 1;  // Identity
        
        while (p > 0) {
            if ((p & 1) == 1) result = matMul(result, A);
            A = matMul(A, A);
            p >>= 1;
        }
        return result;
    }
    
    // Fibonacci using matrix exponentiation - O(log n)
    static long fib(long n) {
        if (n <= 1) return n;
        long[][] base = {{1, 1}, {1, 0}};
        long[][] result = matPow(base, n - 1);
        return result[0][0];
    }
    
    // ==================== FAST I/O ====================
    static String next() throws IOException {
        while (st == null || !st.hasMoreTokens())
            st = new StringTokenizer(br.readLine());
        return st.nextToken();
    }
    
    static int nextInt() throws IOException { return Integer.parseInt(next()); }
    static long nextLong() throws IOException { return Long.parseLong(next()); }
}

File: Templates/cpp/MathTemplate.cpp
/*
 * Author  : Aritra Dutta
 * Target  : Codeforces Expert / CSES
 * 
 * MATH Template - Number Theory, Combinatorics, Primes
 */
#include <bits/stdc++.h>
using namespace std;

using ll = long long;
using vll = vector<ll>;
using vi = vector<int>;

constexpr ll MOD = 1e9 + 7;
constexpr ll MOD2 = 998244353;
constexpr int MAXN = 2e6 + 5;

#define rep(i, a, b) for (int i = (a); i < (b); ++i)

// ==================== Modular Arithmetic ====================
ll mod_add(ll a, ll b, ll m = MOD) { return ((a % m) + (b % m)) % m; }
ll mod_sub(ll a, ll b, ll m = MOD) { return ((a % m) - (b % m) + m) % m; }
ll mod_mul(ll a, ll b, ll m = MOD) { return ((a % m) * (b % m)) % m; }

ll mod_pow(ll x, ll y, ll m = MOD) {
    ll res = 1; x %= m;
    while (y > 0) {
        if (y & 1) res = res * x % m;
        x = x * x % m; y >>= 1;
    }
    return res;
}

ll mod_inv(ll x, ll m = MOD) { return mod_pow(x, m - 2, m); }

// ==================== Factorials & nCr ====================
ll fact[MAXN], inv_fact[MAXN];

void precompute_factorials(int n = MAXN - 1) {
    fact[0] = 1;
    rep(i, 1, n + 1) fact[i] = mod_mul(fact[i-1], i);
    inv_fact[n] = mod_inv(fact[n]);
    for (int i = n - 1; i >= 0; i--) inv_fact[i] = mod_mul(inv_fact[i+1], i+1);
}

ll nCr(int n, int r) {
    if (r < 0 || r > n) return 0;
    return mod_mul(fact[n], mod_mul(inv_fact[r], inv_fact[n-r]));
}

ll nPr(int n, int r) {
    if (r < 0 || r > n) return 0;
    return mod_mul(fact[n], inv_fact[n-r]);
}

// ==================== GCD, LCM, Extended GCD ====================
ll gcd(ll a, ll b) { return b ? gcd(b, a % b) : a; }
ll lcm(ll a, ll b) { return a / gcd(a, b) * b; }

// Returns {gcd, x, y} such that ax + by = gcd(a, b)
tuple<ll, ll, ll> extgcd(ll a, ll b) {
    if (b == 0) return {a, 1, 0};
    auto [g, x, y] = extgcd(b, a % b);
    return {g, y, x - (a / b) * y};
}

// ==================== Sieve of Eratosthenes ====================
vi sieve(int n) {
    vi is_prime(n + 1, 1), primes;
    is_prime[0] = is_prime[1] = 0;
    for (int i = 2; i <= n; i++) {
        if (is_prime[i]) {
            primes.push_back(i);
            for (ll j = (ll)i * i; j <= n; j += i) is_prime[j] = 0;
        }
    }
    return primes;
}

// ==================== Smallest Prime Factor (SPF) ====================
vi spf;
void compute_spf(int n) {
    spf.assign(n + 1, 0);
    iota(spf.begin(), spf.end(), 0);
    for (int i = 2; i * i <= n; i++) {
        if (spf[i] == i) {
            for (int j = i * i; j <= n; j += i) {
                if (spf[j] == j) spf[j] = i;
            }
        }
    }
}

// Prime factorization using SPF - O(log n)
vector<pair<int, int>> factorize(int n) {
    vector<pair<int, int>> factors;
    while (n > 1) {
        int p = spf[n], cnt = 0;
        while (n % p == 0) { n /= p; cnt++; }
        factors.push_back({p, cnt});
    }
    return factors;
}

// ==================== Euler's Totient ====================
ll phi(ll n) {
    ll result = n;
    for (ll i = 2; i * i <= n; i++) {
        if (n % i == 0) {
            while (n % i == 0) n /= i;
            result -= result / i;
        }
    }
    if (n > 1) result -= result / n;
    return result;
}

// ==================== Matrix Exponentiation ====================
using Matrix = vector<vll>;

Matrix multiply(const Matrix& A, const Matrix& B, ll m = MOD) {
    int n = A.size();
    Matrix C(n, vll(n, 0));
    rep(i, 0, n) rep(k, 0, n) rep(j, 0, n) {
        C[i][j] = mod_add(C[i][j], mod_mul(A[i][k], B[k][j], m), m);
    }
    return C;
}

Matrix matrix_pow(Matrix A, ll p, ll m = MOD) {
    int n = A.size();
    Matrix result(n, vll(n, 0));
    rep(i, 0, n) result[i][i] = 1;  // Identity
    while (p > 0) {
        if (p & 1) result = multiply(result, A, m);
        A = multiply(A, A, m);
        p >>= 1;
    }
    return result;
}

// Fibonacci using matrix exponentiation - O(log n)
ll fib(ll n, ll m = MOD) {
    if (n <= 1) return n;
    Matrix base = {{1, 1}, {1, 0}};
    Matrix result = matrix_pow(base, n - 1, m);
    return result[0][0];
}

void solve() {
    precompute_factorials();
    
    int n, r;
    cin >> n >> r;
    cout << nCr(n, r) << '\n';
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    
    int t = 1;
    // cin >> t;
    while (t--) solve();
    
    return 0;
}
